### PR TITLE
Use `[[maybe_unused]]` instead of `(void)` cast

### PR DESCRIPTION
**Pull request by liyli0126**

### CODE REVIEW

File: source/base/function_lib.cc [hunk 1]
```diff
@@ -2015,7 +2015,7 @@ namespace Functions
   FourierSineFunction<dim>::laplacian(const Point<dim>  &p,
                                       const unsigned int component) const
```
Comment (tamiko) [inline source/base/function_lib.cc:2016]: You need to add the `[[maybe_unused]]` attribute in the parameter list.

File: source/base/function_lib.cc [hunk 2]
```diff
@@ -2015,7 +2015,7 @@ namespace Functions
...
   {
-    (void)component;
+    [[mabe_unused]]component;
```
Comment (tamiko) [inline source/base/function_lib.cc:2018]: And simply remove the `(void)component` line.

**Review by tamiko — CHANGES_REQUESTED** [review]

**Comment by liyli0126** [comment]
I replaced a void cast with the more modern [[maybe_unused]]

This is working towards fixing #17247

File: source/base/function_lib.cc [hunk 3]
```diff
@@ -1941,7 +1941,7 @@ namespace Functions
   FourierCosineFunction<dim>::value(const Point<dim>  &p,
                                     const unsigned int component) const
   {
-    (void)component;
+    [[mabe_unused]]component;
```
Comment (bangerth) [inline source/base/function_lib.cc:1944]: This is the wrong place to add the attribute. You want to attach the attribute to the place where the variable is *declared*. So this should look as follows instead:
```
FourierCosineFunction<dim>::value(const Point<dim>  &p,
                                    [[maybe_unused]] const unsigned int component) const
  {
```
Can you change that here and elsewhere?

**Pull request closed without merging**